scripts/scil_clean_qbx_clusters.py

```diff
@@ -0,0 +1,285 @@
+                                          linewidth=background_linewidth)
+    curr_streamlines_actor = actor.line(sft_accepted_on_size[0].streamlines,
+                                        opacity=0.8,
+                                        linewidth=clusters_linewidth)
+
+    scene = window.Scene()
+    interactor_style = interactor.CustomInteractorStyle()
+    show_manager = window.ShowManager(scene, size=(800, 800),
+                                      reset_camera=False,
+                                      interactor_style=interactor_style)
+    scene.add(concat_streamlines_actor)
+    scene.add(curr_streamlines_actor)
+    interactor_style.AddObserver('KeyPressEvent', keypress_callback)
+
+    # Lauch rendering and selection procedure
+    choices, accepted_streamlines, rejected_streamlines = [], [], []
+    show_curr_actor = True
+    show_manager.start()
+
+    # Early exit means everything else is rejected
+    output_accepted_streamlines, output_rejected_streamlines = [], []
+    missing = len(args.in_bundles) - len(choices) - len(sft_rejected_on_size)
+    len_accepted = len(sft_accepted_on_size)
+    rejected_streamlines.extend(range(len_accepted - missing,
+                                      len_accepted))
+    if missing > 0:
+        logging.info('%s clusters automatically rejected from early exit',
+                     missing)
+
+    # Save accepted clusters (by GUI)
+    for idx in accepted_streamlines:
+        streamlines = sft_accepted_on_size[idx].streamlines
+        output_accepted_streamlines.extend(streamlines)
+
+        if args.out_accepted_dir:
+            tmp_sft = StatefulTractogram(streamlines,
+                                         sft_accepted_on_size[0],
+                                         Space.RASMM)
+            tmp_filename = os.path.join(args.out_accepted_dir,
+                                        filename_accepted_on_size[idx])
+            save_tractogram(tmp_sft, tmp_filename, bbox_valid_check=False)
+
+    accepted_sft = StatefulTractogram(output_accepted_streamlines,
+                                      sft_accepted_on_size[0],
+                                      Space.RASMM)
+    save_tractogram(accepted_sft, args.out_accepted, bbox_valid_check=False)
+
+    # Save rejected clusters (by GUI)
+    for idx in rejected_streamlines:
+        streamlines = sft_accepted_on_size[idx].streamlines
+        output_rejected_streamlines.extend(streamlines)
+
+        if args.out_rejected_dir:
+            tmp_sft = StatefulTractogram(streamlines,
+                                         sft_accepted_on_size[0],
+                                         Space.RASMM)
+            tmp_filename = os.path.join(args.out_rejected_dir,
+                                        filename_accepted_on_size[idx])
+            save_tractogram(tmp_sft, tmp_filename, bbox_valid_check=False)
+
+    # Save rejected clusters (by size)
+    for idx in range(len(sft_rejected_on_size)):
```

Comment:
Triple duplication of code. Is extracting in a method (in the script itself or in scilpy library) interesting for other uses ?

Reply:
Since it is really specific with a saving directory and a basename per category I will do a function in that script only